<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>800</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="PointTableWidget" name="point_table" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>1</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QTimeEdit" name="time_edit">
      <property name="frame">
       <bool>false</bool>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
      <property name="buttonSymbols">
       <enum>QAbstractSpinBox::NoButtons</enum>
      </property>
      <property name="displayFormat">
       <string>mm:ss.zzz</string>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QWidget" name="variable_container" native="true">
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QCheckBox" name="auto_move_check_box">
         <property name="text">
          <string>auto move</string>
         </property>
         <property name="checked">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item>
        <widget class="Line" name="line">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label">
         <property name="text">
          <string>update T:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="update_t_spin_box">
         <property name="suffix">
          <string>ms</string>
         </property>
         <property name="maximum">
          <number>5000</number>
         </property>
         <property name="singleStep">
          <number>10</number>
         </property>
         <property name="value">
          <number>500</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="Line" name="line_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>predict T:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="predict_t_spin_box">
         <property name="suffix">
          <string>ms</string>
         </property>
         <property name="maximum">
          <number>5000</number>
         </property>
         <property name="singleStep">
          <number>10</number>
         </property>
         <property name="value">
          <number>100</number>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_2">
      <item>
       <widget class="QPushButton" name="start_button">
        <property name="text">
         <string>start</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="stop_button">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="text">
         <string>stop</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="reset_button">
        <property name="enabled">
         <bool>true</bool>
        </property>
        <property name="text">
         <string>reset</string>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>933</width>
     <height>28</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="replay_action">
   <property name="text">
    <string>replay</string>
   </property>
  </action>
  <action name="save_action">
   <property name="text">
    <string>save</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PointTableWidget</class>
   <extends>QWidget</extends>
   <header location="global">QWidget</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>start_button</tabstop>
  <tabstop>stop_button</tabstop>
  <tabstop>reset_button</tabstop>
  <tabstop>auto_move_check_box</tabstop>
  <tabstop>update_t_spin_box</tabstop>
  <tabstop>predict_t_spin_box</tabstop>
  <tabstop>time_edit</tabstop>
 </tabstops>
 <resources/>
 <connections/>
</ui>
